logstash-core/lib/logstash/util/loggable.rb

```diff
@@ -0,0 +1,33 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "cabin"
+
+module LogStash module Util
+  module Loggable
+    class << self
+      def logger=(new_logger)
+        @logger = new_logger
+      end
+
+      def logger
+        @logger ||= Cabin::Channel.get(LogStash)
+      end
+    end
+
+    def self.included(base)
+      class << base
+        def logger
+          @logger || Loggable.logger
+        end
+
+        def logger=(logger)
```

Comment:
cosmetic: you use two different method param names for almost the same code, maybe use the same names for the sake of consistency? 
```ruby
def logger=(new_logger)

def logger=(logger)
```